tests/saitests/sai_qos_tests.py

```diff
@@ -1755,23 +1755,29 @@ def runTest(self):
 
         # send packets
         try:
+            # Read the initial watermark as baseline
+            pg_shared_wm_res_bases = sai_thrift_read_pg_shared_watermark(self.client, port_list[src_port_id])
+
             # send packets to fill pg min but not trek into shared pool
             # so if pg min is zero, it directly treks into shared pool by 1
             # this is the case for lossy traffic
-            send_packet(self, src_port_id, pkt, pkts_num_leak_out + pkts_num_fill_min)
+            send_packet(self, src_port_id, pkt, pkts_num_leak_out + pkts_num_fill_min - margin)
```

Comment:
We shouldn't subtract the margin here, the reason being the check in L1770, is meant to check that after we have sent all leak out + pg min packets, we shouldn't see shared watermark counter getting incremented since the packets do not occupy the shared buffer

Reply:
Updated logic as per discussion.